core/src/main/java/edu/wpi/grip/core/events/RunPipelineEvent.java

```diff
@@ -0,0 +1,16 @@
+package edu.wpi.grip.core.events;
+
+
+/**
+ * Any event that indicates to the {@link edu.wpi.grip.core.PipelineRunner} that it should run.
+ */
+public interface RunPipelineEvent {
+
+    /**
+     * Indicates to the {@link edu.wpi.grip.core.PipelineRunner} that there is an update to one of the values and it should run again.
+     * @return true if the pipeline should be released to run because of the update
+     */
+    default boolean pipelineShouldRun() {
```

Comment:
Yes, some events have conditional logic regarding whether or not it should run. https://github.com/WPIRoboticsProjects/GRIP/pull/392/files#diff-18bf65d3e77dad8266ea91101e0284feR32